ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueSubmission.scala

```diff
@@ -73,11 +73,13 @@ object KeyValueSubmission {
       sourceDescription: String,
       participantId: ParticipantId): DamlSubmission = {
 
-    val inputDamlState = archives.map(
-      archive =>
-        DamlStateKey.newBuilder
-          .setPackageId(archive.getHash)
-          .build)
+    val inputDamlState =
+      packageUploadDedupKey(participantId, submissionId) ::
```

Comment:
as mentioned below, you might want to instead do repeated calls to `addInputDamlState` (e.g. archive.foreach etc.). this avoids building up these intermediate structures, though of course here this shouldn't really be an issue as we're not uploading thousands of packages.

Reply:
Done